### PR TITLE
Adding live templates for OCaml

### PR DESCRIPTION
**Pull request by QuentinRa**
Adding live templates for OCaml

- begin
- class
- <s>doc</s>
- for
- fun
- func
- if
- let
- lin
- match
- method
- <s>module sig</s> msig
- <s>module struct</s> mstruct
- mtype
- thread
- try
- type
- while
- |
- <s>~f</s>

As explained in #342 . I was quite easy by the way, I mean, running your project 👍 and testing.

### CODE REVIEW

**Comment by QuentinRa** [comment]
What I did to test

- create a file.ml
- writing one of the live templates
- create Test.re
- testing that Reason template is still working

File: src/com/reason/ide/template/OCamlContentType.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+public class OCamlContentType extends TemplateContextType {
+
+    protected OCamlContentType() {
+        super("OCAML", "Ocaml");
+    }
+
+    @Override
+    public boolean isInContext(@NotNull TemplateActionContext templateActionContext) {
+        return templateActionContext.getFile().getName().endsWith(".ml");
```
Comment (giraud) [inline src/com/reason/ide/template/OCamlContentType.java:18]: maybe it could also work with mli interface files

**Comment by giraud** [comment]
Thanks, I really appreciate.

small remarks:
- I prefer to not include `|` and `~f` in the default template, they look too small, too specific
- `module struct` and `module sig` are too complex, I prefer names like mtype, so maybe `mstruct`, `msig`  
- `doc` look very specfic to your needs and I'm not sure it's a good thing as a default

**Comment by QuentinRa** [comment]
Thank you! I have rushed things a bit, sorry this is my first PR.

The operator "|" is used quite a lot (match/type/...), it would save a lot of time. I could remove it, but I don't think this is too specific.

As for ~f, this was for [labels](https://ocaml.org/learn/tutorials/labels.html), but I don't mind removing it.

For module struct/module sig, I pondered a bit to find a suitable name (I picked module struct because they will both show up when writing module, or struct). Well, I will change as you want.

For `doc`, I believe this is [for this](https://github.com/ocaml/odoc).

I took the ones that I had on sublime-text. I will check everything, and make a path 👍.

**Comment by giraud** [comment]
It's just that they are default values that everybody will get, so I prefer to play safe.
That shouldn't be a problem because you can still add your own bindings in addition to these ones.

**Comment by QuentinRa** [comment]
Okay, no problem. I will remove "|" too.

**Comment by QuentinRa** [comment]
Done, I wrote a file in which I listed (and tested) every live template. Maybe, you could use if you want to add something in the documentation (or you can remove it if this is useless).

**Pull request merged**